<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<object-views xmlns="http://axelor.com/xml/ns/object-views"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://axelor.com/xml/ns/object-views
  	https://axelor.com/xml/ns/object-views/object-views_6.0.xsd">

	<grid model="com.axelor.company.db.Sequence" title="Sequences"
		name="sequence-grid">

		<field name="prefix" />
		<field name="padding" />
		<field name="suffix" />
		<field name="nextNumber" />
		<field name="model"/>
	</grid>

	<form model="com.axelor.company.db.Sequence" title="Sequence"
		name="sequence-form" onSave="action-group-next-sequence" >
		<panel>
			<field name="prefix" />
			<field name="padding" />
			<field name="suffix" />
		</panel>
		<panel>
			<field name="nextNumber"  />
			<field name="model" form-view="meta-model-form" grid-view="meta-model-grid"></field>
		</panel>
	</form>
	
	<action-group name="action-group-next-sequence">
		<action name="action-sequence-validate-model-sequence"/>
		<action name="action-sequence-method-generate-sequence"></action>
	</action-group>
	
	<action-validate name="action-sequence-validate-model-sequence" model="com.axelor.company.db.Sequence">
		<error message="can not create sequence for same model again" if="__repo__(Sequence).all().filter('self.model = ?',model?.id).count() > 0"/>
	</action-validate >
	
	<action-method name="action-sequence-method-generate-sequence" model="com.axelor.company.db.Sequence" >
		<call class="com.axelor.company.web.SequenceController" method="generateSequence"/>
	</action-method>


</object-views>